### ISSUE TITLE
Directus "REFRESH_TOKEN_COOKIE_DOMAIN" not working

### ISSUE THREAD

**Issue by Nisthar** [directus/directus#21060]
### Describe the Bug

I am using the latest version of directus using docker. My env variables in docker looks like this
```
      AUTH_PROVIDERS: "google"
      AUTH_GOOGLE_DRIVER: "openid"
      AUTH_GOOGLE_CLIENT_ID: "xxxxx" 
      AUTH_GOOGLE_CLIENT_SECRET: "xxxxx" 
      AUTH_GOOGLE_ISSUER_URL: "https://accounts.google.com"
      AUTH_GOOGLE_IDENTIFIER_KEY: "email"
      AUTH_GOOGLE_ICON: "google"
      AUTH_GOOGLE_LABEL: "Google"
      AUTH_GOOGLE_ALLOW_PUBLIC_REGISTRATION: "true"
      AUTH_GOOGLE_DEFAULT_ROLE_ID: "xxxxxx"
      REFRESH_TOKEN_COOKIE_NAME: "refresh_token"
      REFRESH_TOKEN_COOKIE_DOMAIN: ".localhost"
      REFRESH_TOKEN_COOKIE_SECURE: "true"
      REFRESH_TOKEN_COOKIE_SAME_SITE: "none"
```
I am using nextjs for frontend with https in localhost. I tried using the url `http://localhost:8055/auth/login/google?redirect=https://localhost:3000/` for logging in. I inspect the cookies in the nextjs middleware and i can see the `refresh_token` cookies when the redirect is `https://localhost:3000` but i can't see the cookies if i use the redirect as `https://app.localhost:3000` why not?

TLDR
 I inspect the cookies in the nextjs middleware and i can see the `refresh_token` cookies when the redirect is `https://localhost:3000` but i can't see the cookies if i use the redirect as `https://app.localhost:3000`

### To Reproduce

1. Use the google login provider
2. set "REFRESH_TOKEN_COOKIE_DOMAIN" to ".localhost"
3. Try the login url `http://localhost:8055/auth/login/google?redirect=https://app.localhost:3000/`
4. Can't see any cookies.

### Directus Version

v10.8.3

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by Nisthar** [comment]
am i missing something here?

**Comment by br41nslug** [comment]
`localhost` is somewhat of a special case in browsers and `.localhost` is a reserved TLD which may not function as expected in browsers (cookies are likely ignored on the TLD same as for `.com` or `.net` so cookies dont get overshared).

Try adding a TLD `localhost.dev`/`localhost.com` or using a different domain `test.local`/`directus.test` as cookie domain.

Closing this for now as non-localhost domains seem to work fine.
